File: ai25btech11006/MATGEO/1.11.8/codes/main.c
#include <stdio.h>
#include <math.h>

int main() {
    // Coordinates of points P and Q
    float Px = 4, Py = 3, Pz = -5;
    float Qx = -2, Qy = 1, Qz = -8;
    
    // Direction vector PQ = Q - P
    float dx = Qx - Px;
    float dy = Qy - Py;
    float dz = Qz - Pz;
    
    // Magnitude of PQ
    float magnitude = sqrt(dx*dx + dy*dy + dz*dz);
    
    // Direction cosines (unit vector components)
    float l = dx / magnitude;
    float m = dy / magnitude;
    float n = dz / magnitude;
    
    // Output the result
    printf("Direction Cosines: (%.3f, %.3f, %.3f)\n", l, m, n);
    
    return 0;
}

File: ai25btech11006/MATGEO/1.11.8/codes/main.py
import math

# Coordinates of points P and Q
Px, Py, Pz = 4, 3, -5
Qx, Qy, Qz = -2, 1, -8

# Direction vector PQ = Q - P
dx = Qx - Px
dy = Qy - Py
dz = Qz - Pz

# Magnitude of PQ
magnitude = math.sqrt(dx**2 + dy**2 + dz**2)

# Direction cosines (unit vector components)
l = dx / magnitude
m = dy / magnitude
n = dz / magnitude

# Output the result
print(f"Direction Cosines: ({l:.3f}, {m:.3f}, {n:.3f})")

